### PR TITLE
Configure valve container management

### PR DESCRIPTION
**Pull request by gregswift**

### CODE REVIEW

File: ModuleFile [hunk 1]
```diff
@@ -1,5 +1,5 @@
 name 'flume-repose'
-version '1.0.0'
+version '0.1.1'
```
Comment (gregswift) [inline ModuleFile]: This new version number is lower than the old

File: puppet-module-flume-repose.spec [hunk 2]
```diff
@@ -2,7 +2,7 @@
 %define base_name flume_repose
 
 Name:      puppet-module-%{user}-%{base_name}
-Version:   1.0.0
+Version:   0.1.0
```
Comment (gregswift) [inline puppet-module-flume-repose.spec]: same version problem

File: manifests/valve.pp [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+# [*java_options*]
+# String. Additional java options to pass to java_opts
+# NOTE: this also sets JAVA_OPTS for repose 7+
+# Defaults to <tt>undef</tt>
+#
+# [*saxon_home*]
+# String. Home directory for Saxon. Sets SAXON_HOME
+# Defaults to <tt>undef</tt>
+#
+# [*rh_old_packages*]
```
Comment (gregswift) [inline manifests/valve.pp]: rh_old_package does not appear to be a valid thing in the main class you are passing this to. Remove all references?

File: manifests/valve.pp [hunk 4]
```diff
@@ -0,0 +1,183 @@
...
+# Boolean. At version 6.2 repose renamed several of their packages to
+# standardize between deb/rpm.  This variable exposes access to the old
+# naming on rpm distros. It defaults to <tt>true</tt> for the time being
+# to not break existing users.
+# TODO: Determine a time to default to false. Then when to drop support.
+#
+# [*cfg_new_namespace*]
```
Comment (gregswift) [inline manifests/valve.pp]: cfg_new_namespace  does not appear to be a valid thing in the main class you are passing this to. Remove all references?

**Comment by usnavi** [comment]
@gregswift - does this belong here?  Should it go in https://github.com/rackerlabs/puppet-repose?

File: manifests/valve.pp [hunk 5]
```diff
@@ -0,0 +1,164 @@
+# == Class: repose::valve
```
Comment (gregswift) [inline manifests/valve.pp]: this should be flume_repose::valve

File: manifests/valve.pp [hunk 6]
```diff
@@ -0,0 +1,164 @@
...
+#
+#
+# === Authors
+#
+# * Josh Bell <mailto:josh.bell@rackspace.com>
+# * Alex Schultz <mailto:alex.schultz@rackspace.com>
+# * Greg Swift <mailto:greg.swift@rackspace.com>
+# * c/o Cloud Integration Ops <mailto:cit-ops@rackspace.com>
+#
+class flume-repose::valve (
```
Comment (gregswift) [inline manifests/valve.pp]: ```flume_repose``` (underscore not hyphen is correct syntax)

File: manifests/valve.pp [hunk 7]
```diff
@@ -0,0 +1,164 @@
...
+  $daemon_home       = $flume_repose::params::daemon_home,
+  $log_path          = $flume_repose::params::logdir,
+  $pid_file          = $flume_repose::params::pid_file,
+  $user              = $flume_repose::params::user,
+  $daemonize         = $flume_repose::params::daemonize,
+  $daemonize_opts    = $flume_repose::params::daemonize_opts,
+  $run_opts          = $flume_repose::params::run_opts,
+  $java_options      = undef,
+  $saxon_home        = undef,
+) inherits flume-repose::params {
```
Comment (gregswift) [inline manifests/valve.pp]: ```flume_repose``` (underscore not hyphen is correct syntax)

File: manifests/valve.pp [hunk 8]
```diff
@@ -0,0 +1,164 @@
...
+
+  class { 'flume-repose':
```
Comment (gregswift) [inline manifests/valve.pp]: ```flume_repose``` (underscore not hyphen is correct syntax)

File: manifests/valve.pp [hunk 9]
```diff
@@ -0,0 +1,164 @@
...
+    ensure            => $ensure,
+    enable            => $enable,
+    autoupgrade       => $autoupgrade,
+  }
+
+  file { '/etc/sysconfig/flume-repose':
```
Comment (gregswift) [inline manifests/valve.pp]: hyphen is correct here

File: ModuleFile [hunk 10]
```diff
@@ -1,5 +1,5 @@
 name 'flume-repose'
```
Comment (gregswift) [inline ModuleFile]: ```flume_repose``` (underscore not hyphen is correct syntax)

Comment (gregswift) [inline ModuleFile]: also we should replace the ModuleFile with a metadata.json

File: spec/classes/valve_spec.rb [hunk 11]
```diff
@@ -0,0 +1,86 @@
+require 'spec_helper'
+describe 'flume-repose::valve' do
```
Comment (gregswift) [inline spec/classes/valve_spec.rb]: ```flume_repose``` (underscore not hyphen is correct syntax)

File: puppet-module-flume-repose.spec [hunk 12]
```diff
@@ -2,7 +2,7 @@
 %define base_name flume_repose
```
Comment (gregswift) [inline puppet-module-flume-repose.spec]: we can delete this file too

File: spec/classes/valve_spec.rb [hunk 13]
```diff
@@ -0,0 +1,86 @@
...
+        :ensure => 'absent'
+      } }
+      it {
+        should contain_class('repose').with(
+          'ensure'      => 'absent',
+          'enable'      => 'true',
+          'autoupgrade' => 'false',
+          'container'   => 'valve'
+        )
+        should contain_file('/etc/sysconfig/repose').with_ensure('absent')
```
Comment (gregswift) [inline spec/classes/valve_spec.rb]: ```/etc/sysconfig/flume-repose```

File: spec/classes/valve_spec.rb [hunk 14]
```diff
@@ -0,0 +1,86 @@
...
+
+    context 'with defaults for all parameters' do
+      let(:facts) { { :osfamily => 'RedHat' } }
+      it {
+        should contain_class('repose').with(
+          'ensure'      => 'present',
+          'enable'      => 'true',
+          'autoupgrade' => 'false'
+        )
+        should contain_file('/etc/sysconfig/repose')
```
Comment (gregswift) [inline spec/classes/valve_spec.rb]: ```/etc/sysconfig/flume-repose```

File: spec/classes/valve_spec.rb [hunk 15]
```diff
@@ -0,0 +1,86 @@
...
+          'enable'          => 'true',
+          'autoupgrade'     => 'false',
+          'rh_old_packages' => 'false',
+          'container'       => 'valve'
+        )
+      }
+    end
+
+    context 'with new namespaces' do
+      let(:params) { { :cfg_new_namespace => 'true' } }
```
Comment (gregswift) [inline spec/classes/valve_spec.rb]: i believe this can go away, probably part of why tests are failing

File: spec/classes/valve_spec.rb [hunk 16]
```diff
@@ -0,0 +1,86 @@
...
+          'autoupgrade' => 'false',
+          'container'   => 'valve'
+        )
+        should contain_file('/etc/sysconfig/repose').with_ensure('absent')
+        should_not contain_augeas('repose_sysconfig')
+      }
+    end
+
+    context 'with new packages' do
+      let(:params) { { :rh_old_packages => 'false' } }
```
Comment (gregswift) [inline spec/classes/valve_spec.rb]: i believe this can go away, probably part of why tests are failing

File: spec/classes/valve_spec.rb [hunk 17]
```diff
@@ -0,0 +1,86 @@
...
+      it {
+        should contain_class('repose').with(
+          'ensure'      => 'present',
+          'enable'      => 'true',
+          'autoupgrade' => 'false'
+        )
+        should contain_file('/etc/sysconfig/repose')
+        # TODO: add additional details here
+        should contain_augeas('repose_sysconfig').with_context(
+          '/files/etc/sysconfig/repose')
```
Comment (gregswift) [inline spec/classes/valve_spec.rb]: ```/files/etc/sysconfig/flume-repose```

File: manifests/valve.pp [hunk 18]
```diff
@@ -0,0 +1,164 @@
...
+    group   => root,
+    require => [ Package[$flume_repose::params::package] ],
+    notify  => Service[$flume_repose::params::service],
+  }
+
+
+  # default repose valve sysconfig options
+  if $saxon_home { $saxon_ensure = present } else { $saxon_ensure = absent }
+  $defaults = {
+    'target'  => '/etc/sysconfig/flume_repose',
```
Comment (gregswift) [inline manifests/valve.pp]: I believe this should be ``` /files/etc/sysconfig/flume-repose ```

**Comment by gregswift** [comment]
@usnavi to answer your "does this go here?" my answer is I'm not sure.

@bjoshua is the valve module here because flume runs in valve? Thats my assumption from looking at this.

**Comment by bjoshua** [comment]
@gregswift - Thanks Greg, I caught several of these while I was trying to clean up tests.  - Yes flume-repose runs in a valve container. It may not need to exist as a separate valve manifest because we aren't managing the package any other way, and may just be part of the config manifest. Thanks for all the feedback.

**Comment by usnavi** [comment]
What's the end goal of this PR?  I've already included a service class https://github.com/rackerlabs/puppet-flume_repose/blob/master/manifests/service.pp to set up the app as a service.  Is this similar to that?

As far as valve goes, this doesn't run in the same jvm as valve.  This is essentially a module for flume, as packaged and released by the repose team, in use with the cloud feeds sink, which they develop.  Should we talk in person?  I'm confused.

**Comment by bjoshua** [comment]
Misdirection by flume config was somewhere unexpected, new relic monitoring is taken care of already. Please close this pull request.

**Pull request closed without merging**